client/src/main/java/org/dst/client/DstDictProxy.java

```
@@ -0,0 +1,144 @@
+package org.dst.client;
+
+import org.dst.exception.DictKeyNotFoundException;
+import org.dst.exception.DstException;
+import org.dst.exception.KeyNotFoundException;
+import org.dst.server.generated.CommonProtocol;
+import org.dst.server.generated.DictProtocol;
+import org.dst.server.service.DstDictService;
+import java.util.HashMap;
+import java.util.Map;
+
+public class DstDictProxy {
+
+  private DstDictService service;
+
+  public DstDictProxy(DstDictService service) {
+    this.service = service;
+  }
+
+  public void put(String key, Map<String, String> dict) {
+    DictProtocol.PutRequest.Builder request =
+        DictProtocol.PutRequest.newBuilder();
+    request.setKey(key);
+    DictProtocol.DstDict.Builder builder =
+        DictProtocol.DstDict.newBuilder();
+    for (Map.Entry<String, String> entry : dict.entrySet()) {
+      builder.addKeys(entry.getKey());
+      builder.addValues(entry.getValue());
+    }
+    request.setDict(builder.build());
+    DictProtocol.PutResponse response = service.put(request.build());
+    if (response.getStatus() != CommonProtocol.Status.OK) {
+      throw new KeyNotFoundException(key);
```

throw key not found exception only if the status is key not found.

---

And btw, put should throw key not found, right?